src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/components/collection-select/main.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of the Sulu CMF.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(function() {
+
+    'use strict';
+
+    var defaults = {
+            instanceName: 'collection-select'
+        },
+
+        templates = {
+            toggler: [
+                '<div id="show-ghost-pages"></div>',
+                '<label class="inline spacing-left" for="show-ghost-pages"><%= label %></label>'
+            ].join(''),
+
+            columnNavigation: function() {
+                return [
+                    '<div id="child-column-navigation"/>',
+                    '<div id="wait-container" style="margin-top: 50px; margin-bottom: 200px; display: none;"></div>'
+                ].join('');
+            }
+        },
+
+        namespace = 'sulu.media.collection-select.',
+
+        /**
+         * Open overlay
+         * @event sulu.media.collection-select.open
+         */
+        OPEN = function() {
+            return createEventName.call(this, 'open');
+        },
+
+        /**
+         * Close overlay
+         * @event sulu.media.collection-select.close
+         */
+        CLOSE = function() {
+            return createEventName.call(this, 'close');
+        },
+
+        /**
+         * Selected
+         * @event sulu.media.collection-select.selected
+         */
+        SELECTED = function() {
+            return createEventName.call(this, 'selected');
+        },
+
+        /** returns normalized event names */
+        createEventName = function(postFix) {
+            return namespace + this.options.instanceName + '.' + postFix;
```

Comment:
the instanceName has a default value in this component which i use as instance-names for sub components ...